bitwarden_license/test/Commercial.Core.Test/SecretManagerFeatures/Secrets/UpdateSecretCommandTests.cs

```diff
@@ -0,0 +1,132 @@
+            var secretUpdate = new Secret()
+            {
+                OrganizationId = updatedOrgId,
+                Id = existingSecret.Id,
+                Key = existingSecret.Key,
+            };
+
+            var result = await sutProvider.Sut.UpdateAsync(secretUpdate);
+
+            Assert.Equal(existingSecret.OrganizationId, result.OrganizationId);
+            Assert.NotEqual(existingSecret.OrganizationId, updatedOrgId);
+        }
+
+        [Theory]
+        [BitAutoData]
+        public async Task UpdateAsync_DoesNotModifyCreationDate(Secret existingSecret, SutProvider<UpdateSecretCommand> sutProvider)
+        {
+            sutProvider.GetDependency<ISecretRepository>().GetByIdAsync(existingSecret.Id).Returns(existingSecret);
+
+            var updatedCreationDate = DateTime.UtcNow;
+            var secretUpdate = new Secret()
+            {
+                CreationDate = updatedCreationDate,
+                Id = existingSecret.Id,
+                Key = existingSecret.Key,
+            };
+
+            var result = await sutProvider.Sut.UpdateAsync(secretUpdate);
+
+            Assert.Equal(existingSecret.CreationDate, result.CreationDate);
+            Assert.NotEqual(existingSecret.CreationDate, updatedCreationDate);
+        }
+
+        [Theory]
+        [BitAutoData]
+        public async Task UpdateAsync_DoesNotModifyDeletionDate(Secret existingSecret, SutProvider<UpdateSecretCommand> sutProvider)
+        {
+            sutProvider.GetDependency<ISecretRepository>().GetByIdAsync(existingSecret.Id).Returns(existingSecret);
+
+            var updatedDeletionDate = DateTime.UtcNow;
+            var secretUpdate = new Secret()
+            {
+                DeletedDate = updatedDeletionDate,
+                Id = existingSecret.Id,
+                Key = existingSecret.Key,
+            };
+
+            var result = await sutProvider.Sut.UpdateAsync(secretUpdate);
+
+            Assert.Equal(existingSecret.DeletedDate, result.DeletedDate);
+            Assert.NotEqual(existingSecret.DeletedDate, updatedDeletionDate);
+        }
+
+
+        [Theory]
+        [BitAutoData]
+        public async Task UpdateAsync_RevisionDateIsUpdatedToUtcNow(Secret existingSecret, SutProvider<UpdateSecretCommand> sutProvider)
+        {
+            sutProvider.GetDependency<ISecretRepository>().GetByIdAsync(existingSecret.Id).Returns(existingSecret);
+
+            var updatedRevisionDate = DateTime.UtcNow.AddDays(10);
+            var secretUpdate = new Secret()
+            {
+                RevisionDate = updatedRevisionDate,
+                Id = existingSecret.Id,
+                Key = existingSecret.Key,
+            };
+
+            var result = await sutProvider.Sut.UpdateAsync(secretUpdate);
+
+            Assert.NotEqual(existingSecret.RevisionDate, result.RevisionDate);
+            Assert.True(result.RevisionDate < updatedRevisionDate);
+            Assert.True(DateTime.UtcNow.AddMinutes(-1) < result.RevisionDate);
```

Comment:
We have a `AssertHelper.AssertRecent` for just this reason 😄

Reply:
Very helpful! 

Resolved in https://github.com/bitwarden/server/pull/2201/commits/8104854ca81795a40cfd44c623d23746d2fccf99